src/Microsoft.Framework.Runtime/Project.cs

```diff
@@ -436,9 +476,13 @@ private static IEnumerable<string> GetSourcesSplit(string sourceDescription)
             {
                 foreach (var dependency in dependencies)
                 {
-                    if (String.IsNullOrEmpty(dependency.Key))
+                    if (string.IsNullOrEmpty(dependency.Key))
                     {
-                        throw new InvalidDataException("Unable to resolve dependency ''.");
+
+                        throw ProjectFormatException.Create(
+                            "Unable to resolve dependency ''.",
```

Comment:
Or say that the dependency isn't specified? Or that dependency names can't be empty?